tools/interactive_marker_teleop/src/interactive_marker_teleop.cpp

```diff
@@ -0,0 +1,299 @@
+  InteractiveMarkerControl control;
+  control.always_visible = true;
+  control.markers.push_back(makeMarker("body"));
+  control.markers.push_back(makeMarker("pmc_left"));
+  control.markers.push_back(makeMarker("pmc_right"));
+  msg.controls.push_back(control);
+
+  return msg.controls.back();
+}
+
+void sendMobilityCommand(std::string command, const geometry_msgs::Pose& desired_pose) {
+  if (command == ff_msgs::CommandConstants::CMD_NAME_SIMPLE_MOVE6DOF) {
+    // Make ros command message to send to the executive. see `simple_move.cc`
+    ff_msgs::CommandStamped move_cmd;
+    move_cmd.header.stamp = ros::Time::now();
+    move_cmd.cmd_name = ff_msgs::CommandConstants::CMD_NAME_SIMPLE_MOVE6DOF;
+    move_cmd.cmd_id = "interactive_marker" + std::to_string(move_cmd.header.stamp.sec);
+    move_cmd.cmd_src = "interactive_marker";
+    move_cmd.subsys_name = "Astrobee";
+
+    // Move command has 4 arguements; frame, xyz, xyz tolerance, and rotation
+    move_cmd.args.resize(4);
+    move_cmd.args[0].data_type = ff_msgs::CommandArg::DATA_TYPE_STRING;
+    move_cmd.args[0].s = "world";
+
+    // Set location where you want Astrobee to go to
+    move_cmd.args[1].data_type = ff_msgs::CommandArg::DATA_TYPE_VEC3d;
+    move_cmd.args[1].vec3d[0] = desired_pose.position.x;  // x
+    move_cmd.args[1].vec3d[1] = desired_pose.position.y;  // y
+    move_cmd.args[1].vec3d[2] = desired_pose.position.z;  // z
+
+    // "Tolerance not used!"
+    move_cmd.args[2].data_type = ff_msgs::CommandArg::DATA_TYPE_VEC3d;
+    move_cmd.args[2].vec3d[0] = 0;
+    move_cmd.args[2].vec3d[1] = 0;
+    move_cmd.args[2].vec3d[2] = 0;
+
+    // Target attitude, quaternion, only the first 4 values are used
+    move_cmd.args[3].data_type = ff_msgs::CommandArg::DATA_TYPE_MAT33f;
+    move_cmd.args[3].mat33f[0] = desired_pose.orientation.x;
+    move_cmd.args[3].mat33f[1] = desired_pose.orientation.y;
+    move_cmd.args[3].mat33f[2] = desired_pose.orientation.z;
+    move_cmd.args[3].mat33f[3] = desired_pose.orientation.w;
+    move_cmd.args[3].mat33f[4] = 0;
+    move_cmd.args[3].mat33f[5] = 0;
+    move_cmd.args[3].mat33f[6] = 0;
+    move_cmd.args[3].mat33f[7] = 0;
+    move_cmd.args[3].mat33f[8] = 0;
+
+    // Send command
+    cmd_publisher.publish(move_cmd);
+  } else {
+    ROS_ERROR("only simplemove supported with pose");
+  }
+}
+
+void sendMobilityCommand(std::string command) {
+  ff_msgs::CommandStamped cmd;
+  cmd.header.stamp = ros::Time::now();
+  cmd.subsys_name = "Astrobee";
+  cmd.cmd_name = command;
+  cmd.cmd_id = command;
+
+  if (command == ff_msgs::CommandConstants::CMD_NAME_DOCK) {
+    // Dock has one argument
+    cmd.args.resize(1);
+    cmd.args[0].data_type = ff_msgs::CommandArg::DATA_TYPE_INT;
+    cmd.args[0].i = 1;  // TODO(jdekarske) support other berth
+  }
+
+  cmd_publisher.publish(cmd);
+}
+
+void processFeedback(const InteractiveMarkerFeedbackConstPtr& feedback) {
+  switch (feedback->event_type) {
```

Comment:
I think this could just be an if statement since only one case is checked for

Reply:
Still there because of a TODO, see https://github.com/nasa/astrobee/pull/346#discussion_r776480195